docs/developer-docs/multi-chain/bitcoin/using-btc/runes.mdx

```diff
@@ -0,0 +1,108 @@
+
+## Runes on ICP
+
+Canisters deployed on ICP can sign and submit transactions directly to the Bitcoin network through ICP's [Bitcoin integration](../overview.mdx) and [threshold signatures](/docs/current/references/t-sigs-how-it-works). Specifically, for a canister to etch and mint Runes, it must sign transactions using [threshold Schnorr signatures](/docs/current/references/t-sigs-how-it-works#deuterium-schnorr-release-beta) and have a [Bitcoin taproot address](/docs/current/developer-docs/multi-chain/bitcoin/using-btc/generate-addresses#generating-addresses-with-threshold-schnorr).
+
+## Etching and minting Runes
+
+To etch a Rune through a canister on ICP, first you will need to call the `schnorr_public_key` API to obtain a Schnorr public key. This public key will be used to generate a Bitcoin taproot (P2TR) address for your canister. A taproot address is required to sign and submit Rune transactions, both etching and minting, on Bitcoin. [Learn more about Bitcoin taproot addresses](/docs/current/developer-docs/multi-chain/bitcoin/using-btc/generate-addresses#generating-addresses-with-threshold-schnorr).
+
+Once you have generated a taproot address for your canister, you will need to write a canister method that creates an etching transaction, then signs that transaction with your canister's Schnorr public key. Here's an example of an etching transaction written in Rust:
+
+```rust
+fn set_beginning_block() {
+  let hash = BlockHash::from_str(FIRST_BLOCK_HASH).expect("valid hash");
+  crate::increase_height(FIRST_HEIGHT, hash);
+}
+
+pub(crate) fn init_rune() {
+  set_beginning_block();
+  let rune = Rune(2055900680524219742);
+
+  let id = RuneId { block: 1, tx: 0 };
+  let etching = Txid::all_zeros();
+
+  rune_to_rune_id(|r| r.insert(rune.store(), id)).expect("MemoryOverflow");
+
+  rune_id_to_rune_entry(|r| {
+    r.insert(
+      id,
+      RuneEntry {
+        block: id.block,
+        burned: 0,
+        divisibility: 0,
+        etching,
+        terms: Some(Terms {
+          amount: Some(1),
+          cap: Some(u128::MAX),
+          height: (
+            Some((SUBSIDY_HALVING_INTERVAL * 4).into()),
+            Some((SUBSIDY_HALVING_INTERVAL * 5).into()),
+          ),
+          offset: (None, None),
+        }),
+        mints: 0,
+        premine: 0,
+        spaced_rune: SpacedRune { rune, spacers: 128 },
+        symbol: Some('\u{29C9}'),
+        timestamp: 0,
+        turbo: true,
+      },
+    )
+  })
+  .expect("MemoryOverflow");
+
+  transaction_id_to_rune(|t| t.insert(Txid::store(etching), rune.store())).expect("MemoryOverflow");
+}
+```
+
+<a href="https://github.com/octopus-network/ord-canister/blob/a0632ebd1493d604ad3e82185f75170755552ac3/src/index.rs#L19-L62">
+    <div align="center">View the full example.</div>
+</a>
+
+
+## Querying Rune information
+
+To query information about a Rune, you can specify a query call that returns the Rune information for a given UTXO. Here is an example written in Rust:
```

Comment:
You may want to note here that they are using the `ordinals `crate which is not supported by ICP/DFINITY as the management canister does not currently provide this data. 

Ordinals crate:
`https://crates.io/crates/ordinals/0.0.1`

Reply:
Great point, I've added this information